framework/aster-frame/src/arch/x86/kernel/tsc.rs

```diff
@@ -1,17 +1,29 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use core::sync::atomic::AtomicU64;
+use core::{
+    arch::x86_64::_rdtsc,
+    sync::atomic::{AtomicBool, AtomicU64, Ordering},
+};
+use log::info;
+use trapframe::TrapFrame;
 use x86::cpuid::cpuid;
 
-/// The frequency of tsc. The unit is Hz.
-pub(crate) static TSC_FREQ: AtomicU64 = AtomicU64::new(0);
+use crate::{
+    arch::timer::{
+        pit::{self, OperatingMode},
+        TIMER_FREQ, TIMER_IRQ_NUM,
+    },
+    trap::IrqLine,
+};
 
-const TSC_DEADLINE_MODE_SUPPORT: u32 = 1 << 24;
+/// The frequency of TSC(Hz)
+pub(crate) static TSC_FREQ: AtomicU64 = AtomicU64::new(0);
 
-/// Determine if the current system supports tsc_deadline mode.
-pub fn is_tsc_deadline_mode_supported() -> bool {
-    let cpuid = cpuid!(1);
-    (cpuid.ecx & TSC_DEADLINE_MODE_SUPPORT) > 0
+pub fn init_tsc_freq() {
+    let tsc_freq = determine_tsc_freq_via_cpuid()
+        .map_or(determine_tsc_freq_via_pit(), |freq| freq as u64 * 1000);
```

Comment:
So the two methods of `determine_tsc_freq_via_cpuid` and `determine_tsc_freq_via_pit` return frequencies in different units. Is it possible to eliminate this inconsistency? Manually converting the frequency unit here is error prone.